internal/controller/rbac/provider/binding/reconciler.go

```diff
@@ -169,29 +169,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{Requeue: false}, nil
 	}
 
-	l := &corev1.ServiceAccountList{}
-	if err := r.client.List(ctx, l); err != nil {
-		err = errors.Wrap(err, errListSAs)
-		r.record.Event(pr, event.Warning(reasonBind, err))
-		return reconcile.Result{}, err
+	if pr.Spec.ServiceAccountRef == nil {
```

Comment:
do we have unit test for this change?

Reply:
Changed the implementation as per Nic's feedback and tried to cover the new changes with unit tests.